### ISSUE TITLE
Docker image for v1.0.0

### ISSUE THREAD

**Issue by julien-nc** [DocMarty84/koozic#9]
After reading the announce of v1.0.0 release i installed Koozic with docker and thought i was using the 1.0.0 but it seems that's not the case.

Will you publish v1.0.0 on DockerHub ?

By the way, it was my first time using Koozic, i'm impressed. It's exactly what i was looking for.

I had a few problems though (with 0.8.0). I hope you don't mind that i explain them here as it is maybe solved in 1.0.0.

 * Tracks don't play until the end, the player skips to next track at approximately 80%. Is this a known problem ?
* I can't choose the transcoder used to serve music to the web interface. I put the opus one at the end of the list, it is used anyway. Am i missing something ?

Thanks for this amazing music streamer.

**Comment by DocMarty84** [comment]
Hello,

> Will you publish v1.0.0 on DockerHub ?

Whoops... I was supposed to do it this week (before the announcement was made), I got too busy and forgot about it. Sorry, it should be updated right now.

> Tracks don't play until the end, the player skips to next track at approximately 80%. Is this a known problem ?

Yes, and it drives me crazy. Long story short, modern browser do not load the whole track when playing, only what is necessary. If the tracks lasts too long, the communication between the browser and the server might fail/be closed at some point, and the track stops.

In v1.0.0, there are 2 workarounds:
1. When the tracks ends, KooZic checks if the tracks didn't end prematurely. If it is the case, the playbacks restarts where it stopped. That means a small gap during the playback.
2. Use of Web Audio API (see 'HTML5 or Web Audio' in https://koozic.net/2018/08/10/release-of-v1-0-0/). The track will be fully loaded before being played. That means a potential larger gap between tracks (and it won't be convenient when hitting the 'next' or 'previous' buttons).

In case you still want to stick with HTML5, there is a workaround in Firefox:
1. Go to about:config
2.  Set:
```
media.cache_readahead_limit = 9999
media.cache_resume_threshold = 9999
```

The tracks will start playing instantly, and will also be fully loaded within a few seconds => no connection lost with the server.

> I can't choose the transcoder used to serve music to the web interface. I put the opus one at the end of the list, it is used anyway. Am i missing something ?

Indeed, that won't work. The priority is hardcoded to Opus, Ogg, then MP3:
https://github.com/DocMarty84/oomusic/blob/20736f0829682840fc5f166bbf8157c0c26c33ec/models/oomusic_track.py#L126

Reorganizing the transcoders will only work for a given format. Let's say we have these possible tracks: `[1.opus, 1.ogg, 1.mp3]` And these transcoders:
1. conversion to MP3
2. conversion to ogg
3. conversion to ogg (but with VBR)
4. conversion to opus

The browser will select the first supported format: first Opus, then Ogg and finally MP3. If the browser supports Opus, it will select transcoder 4 since there is only one Opus transcoder, even if it is the last one. If Opus is not supported, it will try Ogg. In this case, it will have the choice between transcoders 2 and 3. It will always take the first one, so in this case transcoder 2. So in this particular situation, the order has an importance. I hope it makes things clearer.

Actually, I didn't think someone would bother with the audio format :-) Opus gives better results than Ogg and MP3 for equivalent bitrates, and is perfectly supported by Firefox and Chrome. Is there a particular reason why you would like to use another codec than Opus?

This is something I could add in the configuration, but I usually avoid unnecessary options in order to keep the interface simple and clean.

**Comment by julien-nc** [comment]
Hi, thanks a lot for your fast an detailed answer !

Thanks for the docker image. I'll try that soon.

> Actually, I didn't think someone would bother with the audio format

I've added a few transcoder entries. I just wanted to be able to quickly change the transcoding quality/codec to make some tests, check if missing ends were caused by some ffmpeg options etc... I'm fine with Opus :wink:.

The way it works is just confusing enough to require a small explanation in the transcoder page IMHO.
